lib/puppet/type/rabbitmq_parameter.rb

```diff
@@ -106,9 +106,6 @@ def validate_value(value)
   def munge_value(value)
     return value if value(:autoconvert) == :false
 
-    value.each do |k, v|
-      value[k] = v.to_i if v =~ %r{\A[-+]?[0-9]+\z}
-    end
-    value
+    value.transform_values { |v| v.match?(%r{\A[-+]?[0-9]+\z}) ? v.to_i : v }
```

Comment:
Or is this better
```suggestion
    value.transform_values do |v|
      if v.match?(%r{\A[-+]?[0-9]+\z})
        v.to_i
      else
        v
      end
    end
```

Reply:
Personally I think that’s a little more readable than the terser syntax but not a strong preference, and don’t claim to be a Ruby expert.

Reply:
Updated